extensions/terminal-suggest/src/completions/upstream/git.ts

```diff
@@ -0,0 +1,9813 @@
+			"tag",
+			"--list",
+			"--sort=-committerdate",
+		],
+		postProcess: function (output) {
+			return output.split("\n").map((tag) => ({
+				name: tag,
+				icon: "🏷️",
+			}));
+		},
+	},
+
+	// Files for staging
+	files_for_staging: {
+		script: ["git", "--no-optional-locks", "status", "--short"],
+		postProcess: (out, context) => {
+			// This whole function is a mess
+
+			const output = filterMessages(out);
+
+			if (output.startsWith("fatal:")) {
+				return [];
+			}
+			let files = output.split("\n").map((file) => {
+				// From "git --no-optional-locks status --short"
+				// M  dev/github.ts // test file that was added
+				//  M dev/kubectl.ts // test file that was not added
+				// A  test2.txt // new added and tracked file
+				// ?? test.txt // new untracked file
+				const alreadyAdded = ["M", "A"].includes(file.charAt(0));
+
+				file = file.trim();
+				const arr = file.split(" ");
+
+				return {
+					working: arr[0],
+					file: arr.slice(1).join(" ").trim(),
+					alreadyAdded,
+				};
+			});
+
+			const paths = output.split("\n").map((file) => {
+				const arr = file
+					.slice(0, file.lastIndexOf("/") + 1)
+					.trim()
+					.split(" ");
+				return arr.slice(1).join(" ").trim();
+			});
+
+			const dirArr = [];
+			if (paths.length >= 2) {
+				let currentDir = paths[0];
+				let count = 1;
+				for (let i = 1; i < paths.length; i++) {
+					if (paths[i].includes(currentDir) && i + 1 !== paths.length) {
+						count++;
+					} else {
+						if (count >= 2) {
+							dirArr.push(currentDir);
+						}
+						count = 1;
+					}
+					currentDir = paths[i];
+				}
+			}
+
+			// Filter out the files that the user has already input in the current edit buffer
+			files = files.filter((item) => {
+				const file = item.file.replace(/^"|"$/g, "");
+				return !context.some((ctx) => {
+					return (
+						ctx === file ||
+						// Need to add support for proper globbing one day
+						(ctx.endsWith("*") && file.startsWith(ctx.slice(0, -1))) ||
+						(ctx.startsWith("*") && file.endsWith(ctx.slice(1)))
+					);
+				});
+			});
+
+			return [
+				...dirArr.map((name) => {
+					return {
+						name: name + "*",
+						description: "Wildcard",
+						icon: "fig://icon?type=asterisk",
+					};
+				}),
+				...files.map((item) => {
+					const file = item.file.replace(/^"|"$/g, "");
+					let ext = "";
+					try {
+						ext = file.split(".").slice(-1)[0];
+					} catch (e) { }
+
+					if (file.endsWith("/")) {
+						ext = "folder";
+					}
+
+					// If the current file is already added
+					// we want to lower the priority
+					const priority = item.alreadyAdded ? 50 : 100;
+					return {
+						name: file,
+						icon: `fig://icon?type=${ext}&color=ff0000&badge=${item.working}`,
+						description: "Changed file",
+						priority,
+					};
+				}),
+			];
+		},
+	},
+
+	getStagedFiles: {
+		script: [
+			"bash",
+			"-c",
+			"git --no-optional-locks status --short | sed -ne '/^M /p' -e '/A /p'",
```

Comment:
There are 3 cases that use `| sed` which will fail currently on Windows. Created https://github.com/microsoft/vscode/issues/240031 to fix and upstream the change